sdk/appcenter-storage/src/main/java/com/microsoft/appcenter/storage/TokenManager.java

```diff
@@ -61,21 +60,22 @@ TokenResult getCachedToken(String partitionName) {
         return getCachedToken(partitionName, false);
     }
 
-    TokenResult getCachedToken(String partitionName, Boolean includeExpiredToken) {
+    TokenResult getCachedToken(String partitionName, boolean includeExpiredToken) {
         TokenResult token = Utils.getGson().fromJson(SharedPreferencesManager.getString(partitionName), TokenResult.class);
-        if (token != null && !includeExpiredToken) {
-            Calendar utcCalendar = Calendar.getInstance(TimeZone.getTimeZone("GMT"));
+        if (token != null) {
+            if (!includeExpiredToken) {
 
-            /* The token is expired. */
-            if (utcCalendar.getTime().compareTo(token.expiresOn()) > 0) {
-                AppCenterLog.warn(LOG_TAG, String.format("Cached token result is expired for partition '%s'", partitionName));
-                return null;
+                /* If the token is expired. */
+                if (System.currentTimeMillis() >= token.expiresOn().getTime()) {

```

Comment:
is it seconds or milliseconds?...

Reply:
nvm seems milli seconds

Reply:
Millis

Reply:
https://developer.android.com/reference/java/util/Date.html#getTime()

Reply:
https://developer.android.com/reference/java/lang/System.html#currentTimeMillis() => already UTC in millis.

Also tested end 2 end and check with debugger on a PST device.